graphdatascience/query_runner/gds_arrow_client.py

```diff
@@ -219,12 +222,33 @@ def handle_flight_error(e: Exception):
             raise e
 
 
-class AuthFactory(ClientMiddlewareFactory):  # type: ignore
-    def __init__(self, middleware: "AuthMiddleware", *args: Any, **kwargs: Any) -> None:
+class UserAgentFactory(ClientMiddlewareFactory):
+    def __init__(self, useragent: str, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._middleware = UserAgentMiddleware(useragent)
+
+    def start_call(self, info: Any) -> ClientMiddleware:
+        return self._middleware
+
+
+class UserAgentMiddleware(ClientMiddleware):
+    def __init__(self, useragent: str, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._useragent = useragent
+
+    def sending_headers(self) -> Dict[str, str]:
+        return {"User-Agent": self._useragent}
```

Comment:
not working yet. Something is overriding this header to `grpc-c++/1.51.1 grpc-c/29.0.0`